internal/controller/controllers/events_uploader_handler.go

```diff
@@ -0,0 +1,304 @@
+			return nil, err
+		}
+		clusterJson, err := json.Marshal(cluster.Cluster)
+		if err != nil {
+			log.WithError(err).Errorf("failed to marshal cluster %s", clusterID)
+			return nil, err
+		}
+		addFile(tw, clusterJson, fmt.Sprintf("%s/events/cluster.json", *clusterID))
+		return cluster, nil
+	}
+	return nil, nil
+}
+
+func infraEnvFile(db *gorm.DB, tw *tar.Writer, infraEnvID *strfmt.UUID, clusterID *strfmt.UUID) error {
+	if infraEnvID != nil {
+		infraEnv, err := common.GetInfraEnvFromDB(db, *infraEnvID)
+		if err != nil {
+			log.WithError(err).Errorf("error getting infraenv %s from db", infraEnvID)
+			return err
+		}
+		iJson, err := json.Marshal(infraEnv)
+		if err != nil {
+			log.WithError(err).Errorf("error marshalling infraenv")
+			return err
+		}
+		// add the file content
+		addFile(tw, iJson, fmt.Sprintf("%s/events/infraenv.json", *clusterID))
+	}
+	return nil
+}
+
+func hostsFile(db *gorm.DB, tw *tar.Writer, cluster *common.Cluster) (*strfmt.UUID, error) {
+	if cluster == nil {
+		return nil, fmt.Errorf("no cluster specified for hosts")
+	}
+	var infraEnvID *strfmt.UUID
+	hosts := cluster.Cluster.Hosts
+	if hosts != nil {
+		hostJson, err := json.MarshalIndent(hosts, "", " ")
+		if err != nil {
+			log.WithError(err).Errorf("error marshalling hosts into file")
+			return nil, err
+		}
+		// add the file content
+		addFile(tw, hostJson, fmt.Sprintf("%s/events/hosts.json", cluster.ID))
+		for _, host := range hosts {
+			if host.InfraEnvID.String() != "" {
+				infraEnvID = &host.InfraEnvID
+			}
+		}
+	}
+	return infraEnvID, nil
+}
+
+func addFile(tw *tar.Writer, contents []byte, fileName string) {
+	// add the file content
+	hdr := &tar.Header{
+		Name: fileName,
+		Mode: 0644,
+		Size: int64(len(contents)),
+	}
+	if err := tw.WriteHeader(hdr); err != nil {
+		log.WithError(err).Errorf("failed writing file header for %s", fileName)
+		return
+	}
+
+	if _, err := tw.Write(contents); err != nil {
+		log.WithError(err).Errorf("failed writing contents to file %s", fileName)
+	}
+}
+
+func (u *uploader) getOCMPullSecret() ([]byte, bool) {
+	if u.client != nil {
+		secret, err := u.client.GetSecret("openshift-config", "pull-secret")
+		if err != nil {
+			err = client.IgnoreNotFound(err)
+			return nil, err != nil
+		}
+		return secret.Data[corev1.DockerConfigJsonKey], true
+	}
+	return nil, false
+}
+
+func (u *uploader) getAuthToken(clusterID *strfmt.UUID) string {
+	data, exists := u.getOCMPullSecret()
+	if !exists {
+		data = getTokenFromCluster(u.db, clusterID)
+	}
+	return ExtractToken(data, openshiftTokenKey)
+}
+
+func getTokenFromCluster(db *gorm.DB, clusterID *strfmt.UUID) []byte {
+	cluster, err := common.GetClusterFromDB(db, *clusterID, common.SkipEagerLoading)
+	if err != nil {
+		log.WithError(err).Errorf("failed to find cluster from db")
+		return nil
+	}
+	return []byte(cluster.PullSecret)
+
+}
+
+func ExtractToken(data []byte, key string) string {
```

Comment:
If this isn't used externally it probably doesn't need to be public.

Reply:
Good point, thank you! Modified it to a private function